android/java/org/chromium/chrome/browser/app/BraveActivity.java

```diff
@@ -957,6 +958,10 @@ public void finishNativeInitialization() {
                         BraveVpnUtils.SUBSCRIPTION_PARAM_TEXT, getPackageName());
             }
         }
+        if (BraveVpnPrefUtils.isSubscriptionPurchase()
+                && !BraveVpnPrefUtils.isLinkSubscriptionDialogShown()) {
+            showLinkVpnSubscriptionDialog();
```

Comment:
Can this also check a feature flag? So we can gate the showing of this dialog on enabling via Griffin?

Also curious - do you have a screen cap of the dialog itself? (maybe you can update PR to include pic)